### PR TITLE
Same as SummaryFormatter, but use STDERR

### PR DESCRIPTION
**Pull request by ko1**

### CODE REVIEW

**Comment by ko1** [comment]
@eregon This is same motivation for the CI system.
https://github.com/ruby/ruby/pull/5487

**Comment by eregon** [comment]
@ko1  `stderrsummary.rb` is not added in this PR

**Comment by ko1** [comment]
OMG!!

**Comment by ko1** [comment]
how about that?

**Comment by eregon** [comment]
I'm a little bit confused.
So if I read this change properly you want to print everything to stdout, except errors/failures and their backtraces which you want to stderr, correct?
So basically you want to consider anything printed on stderr = test errors?

File: spec/mspec/lib/mspec/runner/formatters/base.rb [hunk 1]
```diff
@@ -19,9 +19,9 @@ def initialize(out = nil)
     @count = 0 # For subclasses
 
     if out.nil?
-      @out = $stdout
+      @result_out = @out = $stdout
```
Comment (eregon) [inline spec/mspec/lib/mspec/runner/formatters/base.rb:22]: Could you rename `@result_out` to `@err`?

**Comment by ko1** [comment]
Yes.

You can not see which tests print errors on this page
http://ci.rvm.jp/results/trunk@ruby-iga/3842930

and you need to jump here
http://ci.rvm.jp/logfiles/brlog.trunk.20220221-060059#L471

**Review by eregon — COMMENTED** [review]
I think the functionality makes sense.

Adding a new formatters for this doesn't feel right though.
How about adding a new option like:
```ruby
    on("--error-output", "FILE",
       "Write error output of failing specs to FILE") do |f|
      config[:error_output] = f
    end
```
`'$stderr'` could be special value for `$stderr`, or just use `/dev/stderr` on Linux.

and then add `MSpecOptions.latest` to get the value from inside `BaseFormatter` (without needing a new argument)?

**Comment by eregon** [comment]
And sorry for the late review, I didn't understand initially the purpose of this and forgot about it.

**Comment by ko1** [comment]
I agree option is better approach.
But I can't find out how to introduce new option :p This is why I added new formatter. I'll investigate again.

**Comment by eregon** [comment]
@ko1 I implemented it in https://github.com/ruby/mspec/commit/bd47c2a5998657ad07e610b9873c362b353371cf.

So:
```
$ mspec --error-output stderr core/integer/minus_spec.rb 1>/dev/null
<no output, as expected>
$ mspec --error-output /dev/stderr core/integer/plus_spec.rb 1>/dev/null

1)
Integer#+ fixnum returns self plus the given Integer FAILED
Expected 493 == 4930
to be truthy but was false
/home/eregon/code/rubyspec/core/integer/plus_spec.rb:9:in `block (3 levels) in <top (required)>'
/home/eregon/code/rubyspec/core/integer/plus_spec.rb:4:in `<top (required)>'

2)
Integer#+ bignum returns self plus the given Integer FAILED
Expected 18446744073709551696 == 184467440737095516960
to be truthy but was false
/home/eregon/code/rubyspec/core/integer/plus_spec.rb:32:in `block (3 levels) in <top (required)>'
/home/eregon/code/rubyspec/core/integer/plus_spec.rb:4:in `<top (required)>'
```
also works with the summary formatter if you want to use that:
```
$ mspec --error-output /dev/stderr -f summary core/integer/plus_spec.rb 1>/dev/null

1)
Integer#+ fixnum returns self plus the given Integer FAILED
Expected 493 == 4930
to be truthy but was false
/home/eregon/code/rubyspec/core/integer/plus_spec.rb:9:in `block (3 levels) in <top (required)>'
/home/eregon/code/rubyspec/core/integer/plus_spec.rb:4:in `<top (required)>'

2)
Integer#+ bignum returns self plus the given Integer FAILED
Expected 18446744073709551696 == 184467440737095516960
to be truthy but was false
/home/eregon/code/rubyspec/core/integer/plus_spec.rb:32:in `block (3 levels) in <top (required)>'
/home/eregon/code/rubyspec/core/integer/plus_spec.rb:4:in `<top (required)>'
$ mspec --error-output /dev/stderr -f summary core/integer/minus_spec.rb 1>/dev/null
<no output, as expected>
```

**Comment by ko1** [comment]
Oh, just now, I had overlooked your comment. Thank you!!!

**Pull request closed without merging**